yazi-boot/src/boot.rs

```diff
@@ -130,6 +130,27 @@ impl Boot {
 			);
 		}
 	}
+
+	/// Allows for initialization without forcing the reading of command line
+	/// arguments.
+	pub fn init_with(local_events: HashSet<String>, remote_events: HashSet<String>) -> Boot {
```

Comment:
This initialization process is unnecessary because reading `local_events` and `remote_events` is only necessary when sending messages, which only occurs in the `yazi` program, and `ya` cli doesn't need them.

Reply:
Leaving out the initialization of `BOOT` seems to cause a panic at runtime:

<details><summary>Details</summary>
<p>

```text
$ RUST_BACKTRACE=1 cargo run --bin ya -- sub-static hi,hey,hover,cd
    Finished dev [unoptimized + debuginfo] target(s) in 0.35s
     Running `target/debug/ya sub-static hi,hey,hover,cd`
thread 'tokio-runtime-worker' panicked at /Users/mikavilpas/git/yazi/yazi-shared/src/ro_cell.rs:50:9:
assertion failed: self.is_initialized()
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: <yazi_shared::ro_cell::RoCell<T> as core::ops::deref::Deref>::deref
             at ./yazi-shared/src/ro_cell.rs:50:3
   4: yazi_dds::state::State::load::{{closure}}
             at ./yazi-dds/src/state.rs:81:43
   5: yazi_dds::state::State::load_or_create::{{closure}}
             at ./yazi-dds/src/state.rs:48:18
   6: yazi_dds::client::Client::connect::{{closure}}
             at ./yazi-dds/src/client.rs:161:35
   7: yazi_dds::client::Client::echo_events_to_stdout::{{closure}}
             at ./yazi-dds/src/client.rs:73:61
   8: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/core.rs:328:17
   9: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/loom/std/unsafe_cell.rs:16:9
  10: tokio::runtime::task::core::Core<T,S>::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/core.rs:317:13
  11: tokio::runtime::task::harness::poll_future::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:485:19
  12: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/core/src/panic/unwind_safe.rs:272:9
  13: std::panicking::try::do_call
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/panicking.rs:552:40
  14: ___rust_try
  15: std::panicking::try
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/panicking.rs:516:19
  16: std::panic::catch_unwind
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/panic.rs:142:14
  17: tokio::runtime::task::harness::poll_future
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:473:18
  18: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:208:27
  19: tokio::runtime::task::harness::Harness<T,S>::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:153:15
  20: tokio::runtime::task::raw::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/raw.rs:271:5
  21: tokio::runtime::task::raw::RawTask::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/raw.rs:201:18
  22: tokio::runtime::task::LocalNotified<S>::run
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/mod.rs:427:9
  23: tokio::runtime::scheduler::multi_thread::worker::Context::run_task::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:576:13
  24: tokio::runtime::coop::with_budget
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/coop.rs:107:5
  25: tokio::runtime::coop::budget
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/coop.rs:73:5
  26: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:575:9
  27: tokio::runtime::scheduler::multi_thread::worker::Context::run
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:526:24
  28: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:491:21
  29: tokio::runtime::context::scoped::Scoped<T>::set
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/context/scoped.rs:40:9
  30: tokio::runtime::context::set_scheduler::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/context.rs:176:26
  31: std::thread::local::LocalKey<T>::try_with
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/thread/local.rs:270:16
  32: std::thread::local::LocalKey<T>::with
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/thread/local.rs:246:9
  33: tokio::runtime::context::set_scheduler
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/context.rs:176:9
  34: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:486:9
  35: tokio::runtime::context::runtime::enter_runtime
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/context/runtime.rs:65:16
  36: tokio::runtime::scheduler::multi_thread::worker::run
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:478:5
  37: tokio::runtime::scheduler::multi_thread::worker::Launch::launch::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/scheduler/multi_thread/worker.rs:447:45
  38: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/blocking/task.rs:42:21
  39: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/core.rs:328:17
  40: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/loom/std/unsafe_cell.rs:16:9
  41: tokio::runtime::task::core::Core<T,S>::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/core.rs:317:13
  42: tokio::runtime::task::harness::poll_future::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:485:19
  43: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/core/src/panic/unwind_safe.rs:272:9
  44: std::panicking::try::do_call
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/panicking.rs:552:40
  45: ___rust_try
  46: std::panicking::try
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/panicking.rs:516:19
  47: std::panic::catch_unwind
             at /private/tmp/rust-20240215-5649-scngkd/rustc-1.76.0-src/library/std/src/panic.rs:142:14
  48: tokio::runtime::task::harness::poll_future
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:473:18
  49: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:208:27
  50: tokio::runtime::task::harness::Harness<T,S>::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/harness.rs:153:15
  51: tokio::runtime::task::raw::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/raw.rs:271:5
  52: tokio::runtime::task::raw::RawTask::poll
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/raw.rs:201:18
  53: tokio::runtime::task::UnownedTask<S>::run
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/task/mod.rs:464:9
  54: tokio::runtime::blocking::pool::Task::run
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/blocking/pool.rs:159:9
  55: tokio::runtime::blocking::pool::Inner::run
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/blocking/pool.rs:513:17
  56: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /Users/mikavilpas/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.37.0/src/runtime/blocking/pool.rs:471:13
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</p>
</details> 

Conceptually, I think the DDS doesn't necessarily have to depend on `yazi_boot::Boot` at all, since in this case we're using `yazi-cli`, and in `yazi_boot::Boot` there are many things that are specific to `yazi-fm`. Ideally I think `Boot` could be split, and then this forced initialization could be dropped entirely.

What do you think?

Reply:
This is because the logic of creating a server is entered - a Yazi process can be both a client and a server. When it is found that there are no other running servers, it will create a new one in the `Client::connect()` method. 

The CLI should not go through this creation process; it should only act as a client to connect to an existing server process. So I suggest referring to `Client::shot()` to implement a separate connection for the CLI instead of calling `Client::connect()`.

Reply:
No server is now started. Boot is no longer used.

Reply:
Please remove the `init_with()` since it's no longer used